IMPORT PANDAS

In [1]:
import pandas as pd
import numpy as np

GET DATA

In [2]:
Data_Gojek = pd.read_csv('food_delivery_datasets.csv')

MEMUNCULKAN 5 DATA PERTAMA

In [3]:
Data_Gojek.head()

order_id  voucher_amount                   date_time  eta_seconds  \
0  FOOD-10455               0  2018-08-12T20:35:08.001006         1800   
1  FOOD-10985               0  2018-09-23T17:44:12.243273         2173   
2  FOOD-10911               0  2018-09-19T18:08:09.387991         1418   
3  FOOD-10932               0  2018-09-21T01:24:35.272499         1903   
4  FOOD-10961               0  2018-09-22T17:57:45.523161         1716   

   food_price  cust_id  rating  delivery_fee   resto_id     status  random_id  
0       18900  1089034     5.0          4000  742638033  Completed   0.511149  
1      149000   180146     5.0         11000  742767245  Completed   0.832936  
2       43800   366528     NaN          9000  742618303  Completed   0.828026  
3       39900   511038     5.0         14000  742541172  Completed   0.239377  
4       77700  1220358     NaN          4000  742604875  Completed   0.995587

MENAMPILKAN KOLOM RESTO_ID

In [4]:
id_restoran = Data_Gojek["resto_id"]
id_restoran.head()

0    742638033
1    742767245
2    742618303
3    742541172
4    742604875
Name: resto_id, dtype: int64

MEMBUAT GRUP ID RESTORAN UNTUK MENGETAHUI BERAPA PESANANNYA

In [5]:
Group_restoran = Data_Gojek.groupby("resto_id")

In [6]:
Group_restoran["resto_id"].count()

resto_id
200000639    1
200000964    2
200001006    1
200001973    1
200001979    1
            ..
742808523    1
742809792    1
742810086    1
742813095    2
742814735    1
Name: resto_id, Length: 671, dtype: int64

sorting data untuk total case dengan resto_id terbanyak (descending).

In [7]:
Group_restoran["resto_id"].count() \
                                        .reset_index(name='count') \
                                        .sort_values(['count'], \
                                                    ascending=False)

resto_id  count
311  742591683     12
364  742610474     12
587  742731698      9
390  742622470      9
500  742676756      9
..         ...    ...
250  742565366      1
249  742564676      1
248  742564675      1
247  742562594      1
670  742814735      1

[671 rows x 2 columns]

jadi id_restoran yang memiliki order terbanyak adalah 742591683 dan 742610474 dengan memiliki 12 orderan